src/simtools/simtel/simtel_io_event_writer.py

```diff
@@ -15,144 +18,265 @@
         """Process MC run header and update data lists."""
         mc_head = eventio_object.parse()
         self.n_use = mc_head["n_use"]  # reuse factor n_use needed to extend the values below
         self._logger.info(f"Shower reuse factor: {self.n_use} (viewcone: {mc_head['viewcone']})")
 
-    def _process_mc_shower(self, eventio_object):
+    def _process_file_info(self, file_id, file):
+        """Process file information and append to file info list."""
+        run_info = get_corsika_run_header(file)
+        particle = PrimaryParticle(
+            particle_id_type="eventio_id", particle_id=run_info.get("primary_id", 1)
+        )
+        self.file_info.append(
+            {
+                "file_name": str(file),
+                "file_id": file_id,
+                "particle_id": particle.corsika7_id,
+                "zenith": 90.0 - np.degrees(run_info["direction"][1]),
+                "azimuth": np.degrees(run_info["direction"][0]),
+                "nsb_level": self._get_preliminary_nsb_level(str(file)),
+            }
+        )
+
+    def _process_mc_shower(self, eventio_object, file_id):
         """
         Process MC shower and update shower event list.
 
         Duplicated entries 'self.n_use' times to match the number simulated events with
         different core positions.
         """
-        self.shower = eventio_object.parse()
-
-        self.event_data.simulated_energy.extend([self.shower["energy"]] * self.n_use)
-        self.event_data.shower_azimuth.extend([self.shower["azimuth"]] * self.n_use)
-        self.event_data.shower_altitude.extend([self.shower["altitude"]] * self.n_use)
+        shower = eventio_object.parse()
+
+        self.shower_data.extend(
+            {
+                "shower_id": shower["shower"],
+                "event_id": None,  # filled in _process_mc_event
+                "file_id": file_id,
+                "simulated_energy": shower["energy"],
+                "x_core": None,  # filled in _process_mc_event
+                "y_core": None,  # filled in _process_mc_event
+                "shower_azimuth": shower["azimuth"],
+                "shower_altitude": shower["altitude"],
+                "area_weight": None,  # filled in _process_mc_event
+            }
+            for _ in range(self.n_use)
+        )
 
     def _process_mc_event(self, eventio_object):
-        """Process MC event and update shower event list."""
+        """
+        Process MC event and update shower event list.
+
+        Expected to be called n_use times after _process_shower.
+        """
         event = eventio_object.parse()
 
-        self.event_data.shower_id.append(event["shower_num"])
-        self.event_data.x_core.append(event["xcore"])
-        self.event_data.y_core.append(event["ycore"])
-        self.event_data.area_weight.append(event["aweight"])
+        shower_data_index = len(self.shower_data) - self.n_use + event["event_id"] % 100
```

Comment:
I assume this gives a running index, but not sure about the event_id % 100. Di you check this?

Reply:
Yes, this is how pyeventio implemented the index.

Reply:
yes, this is how pyeventio encodes the shower id.